Gems/ROS2/Code/Source/RobotImporter/SDFormat/Hooks/ROS2SkidSteeringModelHook.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <ROS2/Frame/ROS2FrameComponent.h>
+#include <RobotControl/Controllers/SkidSteeringController/SkidSteeringControlComponent.h>
+#include <RobotControl/ROS2RobotControlComponent.h>
+#include <RobotImporter/SDFormat/ROS2ModelPluginHooks.h>
+#include <RobotImporter/SDFormat/ROS2SDFormatHooksUtils.h>
+#include <VehicleDynamics/ModelComponents/SkidSteeringModelComponent.h>
+#include <VehicleDynamics/Utilities.h>
+
+#include <sdf/Joint.hh>
+#include <sdf/Link.hh>
+#include <sdf/Model.hh>
+#include <sdf/Plugin.hh>
+
+namespace ROS2::SDFormat
+{
+    namespace Parser
+    {
+        VehicleDynamics::VehicleConfiguration CreateVehicleConfiguration(
+            const sdf::ElementPtr element, const sdf::Model& sdfModel, const CreatedEntitiesMap& createdEntities)
+        {
+            VehicleDynamics::VehicleConfiguration configuration;
+            auto addAxle = [&sdfModel, &createdEntities, &configuration](
+                               const std::string& jointNameLeft,
+                               const std::string& jointNameRight,
+                               const float wheelDiameter,
+                               AZStd::string tag) -> void
+            {
+                const auto entityIdLeft = HooksUtils::GetJointEntityId(jointNameLeft, sdfModel, createdEntities);
+                const auto entityIdRight = HooksUtils::GetJointEntityId(jointNameRight, sdfModel, createdEntities);
+                if (entityIdLeft.IsValid() && entityIdRight.IsValid())
+                {
+                    HooksUtils::EnableMotor(entityIdLeft);
+                    HooksUtils::EnableMotor(entityIdRight);
+                    constexpr bool steering = false; // Skid steering model does not have any steering wheels.
+                    constexpr bool drive = true;
+                    configuration.m_axles.emplace_back(VehicleDynamics::Utilities::Create2WheelAxle(
+                        entityIdLeft, entityIdRight, AZStd::move(tag), wheelDiameter / 2.0f, steering, drive));
+                }
+                else
+                {
+                    AZ_Warning(
+                        "CreateVehicleConfiguration",
+                        false,
+                        "Cannot find entity ID for one of the joints: %s or %s",
+                        jointNameLeft.c_str(),
+                        jointNameRight.c_str());
```

Comment:
More detailed warnings is always helpful.

```suggestion
                    AZ_Warning(
                        "CreateVehicleConfiguration",
                        entityIdLeft.IsValid(),
                        "Cannot find entity ID for one of the joint %s",
                        jointNameLeft.c_str());
                    AZ_Warning(
                        "CreateVehicleConfiguration",
                        entityIdRight.IsValid(),
                        "Cannot find entity ID for one of the joints %s",
                        jointNameRight.c_str());
```

Reply:
I will modify the code accordingly in the following PR (Ackermann control), as the automated testing already succeeded in this PR and this PR was on hold for a long time.

Reply:
Pushed in https://github.com/o3de/o3de-extras/pull/642